iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake

```diff
@@ -120,7 +120,7 @@ Macro(iox_make_unique_includedir)
 
     if(MAKE_UNIQUE_INCLUDEDIR)
       set(_unique_dir "iceoryx/v${IOX_VERSION_STRING}")
-      if(PREFIX STREQUAL "")
+      if("${PREFIX}" STREQUAL "")
```

Comment:
Not sure what would be more idiomatic cmake. Your fix or `if(NOT ${PREFIX} OR ${PREFIX} STREQUAL ""`. The fix seems to work but if we for whatever reason rename `PREFIX` and forget to rename it in the string, cmake would not complain and happily. Although it would be the same with `if(NOT ${PREFIX} OR ${PREFIX} STREQUAL ""`.

The issue in this case was that `PREFIX` was used instead of `${PREFIX}`. The latter one will always result in a cmake error if it is used but does not exist. The question is if we really use this variable anywhere and if we could just get rid of it?

Reply:
Okay, I take everything back I wrote above. It seems `${VAR}` and `"${VAR}"` are error prone
https://stackoverflow.com/questions/19982340/cmake-compare-to-empty-string-with-strequal-failed

The most safe solution which also states the intent and prevents our futures selves from removing the quotation marks is probably this
```cmake
if(NOT DEFINED PREFIX OR PREFIX STREQUAL "")
```

Reply:
This is a possible solution bould could also not prevent that someone use `set(PREFIX " ")` to circumvent this.
One idea would be to hardcode the unique include dir directly without setting the Prefix in every library.

What we currently do is that the `iox_add_library` macro take the `PREFIX` as argument for the `PROJECT_PREFIX`
and `INSTALL_INTERFACE`.
Before that we must try to evaluate somehow that PREFIX is valid and set it to the unique include dir (e.g. `iceoryx/v2.90.0`).
We basically make the include dir unique by adding iceoryx and the version number. So why not use the project version of CMake directly?

We could have something like:
```cmake
iox_add_library(
    TARGET                      iceoryx_hoofs
    NAMESPACE                   iceoryx_hoofs
    PROJECT_PREFIX              iceoryx/v${CMAKE_PROJECT_VERSION}
    PUBLIC_LIBS                 iceoryx_hoofs::iceoryx_platform
    ....
    INSTALL_INTERFACE           include/iceoryx/v${CMAKE_PROJECT_VERSION}
```

With that we can ensure that the version is set properly and can get rid of the macro. If the projects version is empty, the CMake will drop an error here.
Would this make sense

Reply:
@dkroenke I like your solution and would go for it!

But when all libraries and executables are always using the same `PROJECT_PREFIX` (`iceoryx/v${CMAKE_PROJECT_VERSION}`) we could also set it directly in `iox_add_library` and `iox_add_executable` and remove the `PROJECT_PREFIX` option from them. Then we have it under our control and the user does not forget it.

Would this work?

Reply:
@elfenpiff as long as `ìox_add_...` isn't used for something outside of iceoryx, this should work

Reply:
This will not work with packages like iceoryx_dds which are planned to be moved outside of iceoryx.
We are already defining a lot centrally in iceoryx_hoofs but we should keep in mind to leave configuration options open for later refactorings.

Reply:
@dkroenke you are right but in this case we do this centrally for a good reason. It would be great if we could find a way to make this work even in iceoryx_dds otherwise we reintroduce all the issues and bugs we already solved with the approach.

But this is a task for the future ...

Reply:
@elBoberido @elfenpiff Reworked the PR by setting PREFIX directly and removing the macro check.